booster/tools/booster/Server.hs

```diff
@@ -173,16 +180,20 @@ main = do
                     Standard -> renderStandardPretty (ExeName "") (TimeSpec 0 0) TimestampsDisable
                     OneLine -> renderOnelinePretty (ExeName "") (TimeSpec 0 0) TimestampsDisable
                     Json -> renderJson (ExeName "") (TimeSpec 0 0) TimestampsDisable
-                koreLogLateFilter = case logFormat of
-                    OneLine ->
-                        if contexLoggingEnabled
-                            then \txt ->
-                                let contextStr = Text.unpack $ extractLogMessageContext txt
-                                 in -- FIXME: likely terrible performance! Use something that does not unpack Text
-                                    not (any (flip Glob.match contextStr) negGlobPatterns)
-                                        && any (flip Glob.match contextStr) globPatterns
-                            else const True
-                    _ -> const True
+                koreLogEarlyFilter = case logFormat of
+                    Json -> \l@(Log.SomeEntry ctxt e) ->
+                        Log.oneLineJson e /= Null && koreFilterContext (ctxt <> [l])
+                    OneLine -> \l@(Log.SomeEntry ctxt _) -> koreFilterContext $ ctxt <> [l]
+                    Standard -> const True
+                koreFilterContext ctxt =
+                    not contexLoggingEnabled
```

Comment:
Looks correct, albeit contrived. Maybe is `if`?

Reply:
it was an if, but hlint didnt like that :D